plugin-server/src/cdp/utils/hog-function-filtering.ts

```diff
@@ -0,0 +1,132 @@
+import { ExecResult } from '@posthog/hogvm'
+import { logger } from '@sentry/utils'
+import { DateTime } from 'luxon'
+import { Histogram } from 'prom-client'
+
+import { UUIDT } from '../../utils/utils'
+import { execHog } from '../services/hog-executor.service'
+import {
+    HogFunctionAppMetric,
+    HogFunctionFilterGlobals,
+    HogFunctionInvocationLogEntry,
+    HogFunctionType,
+} from '../types'
+
+const hogFunctionFilterDuration = new Histogram({
+    name: 'cdp_hog_function_filter_duration_ms',
+    help: 'Processing time for filtering a function',
+    // We have a timeout so we don't need to worry about much more than that
+    buckets: [0, 10, 20, 50, 100, 200],
+    labelNames: ['type'],
+})
+
+interface HogFunctionFilterResult {
+    match: boolean
+    error?: unknown
+    logs: HogFunctionInvocationLogEntry[]
+    metrics: HogFunctionAppMetric[]
+}
+
+/**
+ * Shared utility to check if an event matches the filters of a HogFunction.
+ * Used by both the HogExecutorService (for destinations) and HogTransformerService (for transformations).
+ */
+export function checkHogFunctionFilters(options: {
+    hogFunction: HogFunctionType
```

Comment:
Once you get to 4 or so params its better to just create an options object. Way easier to read as the caller than a bunch of random args in a list